var/spack/repos/builtin/packages/onnx/package.py

```diff
@@ -36,3 +38,8 @@ class Onnx(CMakePackage):
     depends_on('protobuf')
 
     generator = 'Ninja'
+
+    def cmake_args(self):
+        # Try to get ONNX to use the same version of python as the spec is using
+        args = ['-DPY_VERSION={}'.format(self.spec['python'].version.up_to(2))]
```

Comment:
This is not python 2.6 compliant. I suggest the following:

```suggestion
        args = ['-DPY_VERSION={0}'.format(self.spec['python'].version.up_to(2))]
```